packages/klighd-core/src/search-bar/searchbar-panel.tsx

```diff
@@ -0,0 +1,707 @@
+                tooltip.style.borderRadius = '4px'
+                tooltip.style.fontSize = '12px'
+                tooltip.style.display = 'none'
+                document.body.appendChild(tooltip)
+                this.tooltipEl = tooltip
+            }
+        } else {
+            document.removeEventListener('keydown', this.handleEscapeKey)
+            document.removeEventListener('keydown', this.handleKeyNavigation)
+            document.removeEventListener('keydown', this.handleTabKey)
+            document.removeEventListener('keydown', this.handleExitKeycombo)
+            this.actionDispatcher.dispatch(ClearHighlightsAction.create())
+            if (this.tooltipEl) {
+                this.tooltipEl.style.display = 'none'
+            }
+            this.tagInputVisible = false
+            this.clearError()
+        }
+    }
+
+    /**
+     * Update variable searchResults
+     * @param results : an array containing the search result elements
+     */
+    public setResults(results: SModelElement[]): void {
+        this.searchResults = results
+        this.selectedIndex = -1
+        /* Debug: */ console.log("[SearchBar] setting results:", results)
+    }
+
+    /**
+     * Updates the variable textRes.
+     * @param res : an array containing the text field from each search result match
+     */
+    public setTextRes(res: string[]): void {
+        this.textRes = res
+    }
+
+    @inject(TYPES.IActionDispatcher) protected readonly actionDispatcher: IActionDispatcher
+
+    readonly id: 'search-bar-panel'
+    readonly title: 'Search'
+    readonly position: number = 0
+
+    onUpdate(callback: () => void): void {
+        this.updateCallbacks.push(callback)
+    }
+
+    update(): void {
+        for (const callback of this.updateCallbacks) {
+            callback()
+        }
+    }
+
+    /**
+     * Toggles the visibility of the tag input field
+     */
+    private toggleTagInput(): void {
+        this.tagInputVisible = !this.tagInputVisible
+
+        if (!this.tagInputVisible) {
+            const textInput = document.getElementById('search') as HTMLInputElement 
+            const tagInput = document.getElementById('tag-search') as HTMLInputElement
+            if (tagInput) tagInput.value = ''
+            if (textInput) {
+                textInput.value === '' ? this.resetUI() : this.performSearch()
+            }
+        }
+
+        this.update()
+
+        if (this.tagInputVisible) {
+            setTimeout(() => {
+                const tagInput = document.getElementById('tag-search') as HTMLInputElement
+                if (tagInput) {
+                    tagInput.focus()
+                }
+            }, 0)
+        }
+    }
+
+    /**
+     * Performs the search with both main and tag inputs
+     */
+    private performSearch(): void {
+        const mainInput = document.getElementById('search') as HTMLInputElement
+        const tagInput = document.getElementById('tag-search') as HTMLInputElement
```

Comment:
I think this is quite dangerous and inefficient to search the entire document for your IDs you set on your input fields. If something else uses these IDs, this functionality breaks. You should instead use onChange handlers on these input fields and directly update variables in this class.